src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -191,13 +191,13 @@ static Object* pysec_cell(Section* sec) {
 }
 
 static int NpySObj_contains(PyObject* s, PyObject* obj, const char* string) {
-    /* Checks is provided PyObject* s contains obj */
-    if (!PyObject_HasAttrString(obj, string)) {
+    /* Checks is provided PyObject* s matches obj.<string> */
+    auto pyobj = nb::borrow(obj);  // keep refcount+1 during use
```

Comment:
is this `nb::borrow` needed, or could it be `nb::steal`?

Reply:
Note that `steal` should only be used in case we want to acquire the ownership of the object, which is not the case here - we are borrowed the object, as function arguments generally are. If you did so the ref count of the object would be decremented upon return, unless you explicitly `.release()`.

What maybe you want is, since the object is borrowed, we don't need to do any ref count management. For that we create a simple `nb::handle`

Reply:
Ah, yes, nb::handle, good point.

`we are borrowed the object, as function arguments generally are.`
I'm not sure they generally are; from your quote from the "Extending and Embedding the Python Interpreter": 

`When passing an object reference into a function, there is no need to increment the reference count: the function receives ownership of the reference.`

Reply:
Humm, that might not be a good excerpt... To quote more

```
Whenever an object reference is passed into or out of a function, it is part of the function’s interface specification whether ownership is transferred with the reference or not.
...
When you pass an object reference into another function, in general, the function borrows the reference from you 
```

Reply:
I think the important point is that, when the object is passed to a function is it guaranteed we have ref_count > 0. Then it is up to the function to claim ownership, or simply "observe" the reference.

Reply:
> the function borrows the reference from you

Right, but that doesn't mean that one has to use `nb::borrow` in all cases.
It depends on what one is calling, and whether a `strong` reference is needed.
The `nb::borrow` is a bit of a misnomer, since it does a `Py_INCREF()` which is different from how the python docs describe a borrowed reference vs. a strong one (ie: https://docs.python.org/3/glossary.html#term-borrowed-reference vs https://docs.python.org/3/glossary.html#term-strong-reference)

I think this impedance mismatch in the documentation is what is tripping us up.

Reply:
Really good discussion. In the meantime I addressed this in #3024 and indeed it works with the incref